File: PetalBid.Api/Domain/Entities/Auction.cs
using PetalBid.Api.Domain.Enums;
using PetalBid.Api.ValidationAttributes;

namespace PetalBid.Api.Domain.Entities;

public class Auction
{
	public int Id { get; set; }
	public string Description { get; set; } = string.Empty;
	[FutureDate]
	public DateTime StartsAt { get; set; }
	public ClockLocation ClockLocation { get; set; }
	public AuctionStatus Status { get; set; } = AuctionStatus.Pending; // Default status is Pending

	public int? Quantity { get; set; }
	public decimal? ReservePrice { get; set; }
	public int AuctioneerId { get; set; }
	public Auctioneer? Auctioneer { get; set; } = null!;
}

File: client-app/src/types/auction.ts
import { ClockLocation } from './clockLocation';
import { User } from './user';

export enum AuctionStatus {
	Pending = 0,
	Active = 1,
	Paused = 2,
	Ended = 3,
}

export interface CreateAuctionDto {
	description: string;
	startsAt: string; // ISO 8601 string
	clockLocation: ClockLocation;
	auctioneer: User;
}

export interface Auction {
	id: number;
	description: string;
	startsAt: string; // ISO 8601 string
	quantity: number;
	reservePrice: number;
	clockLocation: ClockLocation;
	status: AuctionStatus; // Add auction status
	auctioneer: User;
}